### PR TITLE
Run tests in CI

### PR DESCRIPTION
**Pull request by alathon**
This adds test running as part of CI for pull requests (except those with `nodeploy`).

TODO: Add test running for `develop`/`master` branches too

### Changes
- Add elixir container to build pod used by Jenkinsfile
- Spin up test-only ephemeral PostgreSQL to run tests against
- Add deployment-config/postgresql-ephemeral.yaml so you can set admin
password in container
- Add junit-formatter to apps/metis to output JUnit-style xml from tests
- Collect junit test results after running tests

### To test
1. Run `MIX_ENV=test POSTGRESQL_DATABASE=test mix test` locally
2. Trigger a CI build on OpenShift Jenkins and observe that: (i) it runs tests, (ii) JUnit-style test results are collected under the build in Jenkins

### CODE REVIEW

File: build-pod.yaml [hunk 1]
```diff
@@ -34,6 +41,9 @@ spec:
       command:
         - cat
   volumes:
+    - name: db-data
```
Comment (alathon) [inline build-pod.yaml]: TODO: Remove this - this was a hold-over from when I tried running tests in a build-pod sidecar container.

File: deployment-config/postgresql-ephemeral.yaml [hunk 2]
```diff
@@ -1,127 +1,194 @@
+---
```
Comment (alathon) [inline deployment-config/postgresql-ephemeral.yaml]: As this file is verbatim the same as OpenShift Origin's own template, with the addition of being able to set `POSTGRESQL_ADMIN_PASSWORD`, this is probably a decent thing to do as a PR to the `origin` repo; if they merge that in, we can just use the `openshift//postgresql-ephemeral` template instead.

**Comment by alathon** [comment]
Note: This doesn't run tests on `develop`/`master` yet -- I have yet to add that. Pushing the PR up now anyway, so we can track progress.

**Comment by alathon** [comment]
I also have yet to update the README to reflect these additions.

File: deployment-config/postgresql-ephemeral.yaml [hunk 3]
```diff
@@ -1,127 +1,194 @@
...
+          name: "${DATABASE_SERVICE_NAME}"
+      spec:
+        containers:
+        - capabilities: {}
+          env:
+          - name: POSTGRESQL_USER
+            valueFrom:
+              secretKeyRef:
+                key: database-user
+                name: "${DATABASE_SERVICE_NAME}"
```
Comment (jangnezda) [inline deployment-config/postgresql-ephemeral.yaml]: I'm haven't yet looked at secrets management in Openshift, so I don't know how they are managed. But, I'm getting failed PR deployments with message: `Error: secrets "postgresql" not found`. Probably some minor thing related to env variables.

**Pull request merged**